common/modules/nft_deposit/src/nft_deposit.rs

```diff
@@ -0,0 +1,145 @@
+#![no_std]
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use common_structs::{GenericEsdtAmountPair, Nonce};
+
+#[elrond_wasm_derive::module]
+pub trait NftDepositModule: token_send::TokenSendModule + token_supply::TokenSupplyModule {
+    fn deposit_tokens(
+        &self,
+        payment_token_id: TokenIdentifier,
+        payment_token_nonce: Nonce,
+        payment_amount: Self::BigUint,
+    ) -> SCResult<()> {
+        require!(payment_amount != 0, "Cannot deposit 0 tokens");
+        require!(payment_token_nonce != 0, "Cannot deposit fungible tokens");
+        require!(
+            self.nft_deposit_accepted_token_ids()
+                .contains(&payment_token_id),
+            "Not an accepted token id"
+        );
+
+        let payment = GenericEsdtAmountPair {
+            token_id: payment_token_id,
+            token_nonce: payment_token_nonce,
+            amount: payment_amount,
+        };
+
+        let mut index = 1;
+        let mut entry_updated = false;
+        let caller = self.blockchain().get_caller();
+        let deposit_len = self.nft_deposit(&caller).len();
+
+        while index <= deposit_len {
+            let mut entry = self.nft_deposit(&caller).get(index);
+
+            if self.equal_token_type(&entry, &payment) {
+                entry.amount = &entry.amount + &payment.amount;
+                self.nft_deposit(&caller).set(index, &entry);
+                entry_updated = true;
+                break;
+            }
+
+            index += 1;
+        }
+
+        if !entry_updated {
+            require!(
+                deposit_len + 1 < self.nft_deposit_max_len().get(),
+                "Deposit is full"
+            );
+            self.nft_deposit(&caller).push(&payment);
+        }
+
+        Ok(())
+    }
+
+    #[endpoint(withdrawAllTokensFromDeposit)]
+    fn withdraw_all_tokens_from_deposit(
```

Comment:
It actually don't, just checked. Though it works like a regular vector from Rust, apparently not. Will fix this

Reply:
what does this mean? the `clear_entry` function will not modify the length of the vector?

Reply:
Yes, it will just leave an empty entry and will not modify the vec len. Replaced VecMapper with SingleValueMapper of Vec so we'll no longer have empty entries

Reply:
Resolved